packages/itwin/tree-widget/src/components/trees/common/UseHierarchyVisibility.ts

```diff
@@ -79,15 +79,21 @@ export function useHierarchyVisibility({ visibilityHandlerFactory }: UseHierarch
     const subscription = calculate
       .pipe(
         distinct(undefined, visibilityChanged),
+        observeOn(asyncScheduler),
         mergeMap((node) => defer(async () => ({ node, status: await handler.getVisibilityStatus(node.nodeData) })).pipe(takeUntil(visibilityChanged))),
+        tap({
+          next: ({ node, status }) => {
+            visibilityStatusMap.current.set(node.id, {
+              node,
+              status,
+              needsRefresh: false,
+            });
+          },
+        }),
+        throttleTime(100, undefined, { leading: true, trailing: true }),
       )
       .subscribe({
-        next: ({ node, status }) => {
-          visibilityStatusMap.current.set(node.id, {
-            node,
-            status,
-            needsRefresh: false,
-          });
+        next: () => {
```

Comment:
Do we still need next here?

Reply:
This observable never completes and waits for new nodes that needs visibility determined (when those nodes are rendered) so complete is never called.